### PR TITLE
Update Microsoft.Resources swagger to include new alias contract

### PR DESCRIPTION
**Pull request by eladperets**
These properties are already available for some time now, the swagger is out of date.
The main goal is to add the alias path metadata related properties. I also had to add the default API version and API profiles because they show in the example payload and it won't pass validation otherwise.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json [hunk 1]
```diff
@@ -4581,6 +4586,96 @@
...
+            "String",
+            "Object",
+            "Array",
+            "Integer",
+            "Number",
+            "Boolean"
+          ],
+          "x-ms-enum": {
+            "name": "AliasPathTokenType",
+            "modelAsString": false,
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4617]: this should be true unless you have an explicit need to model this as a native numeric enum in all SDK languages

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json [hunk 2]
```diff
@@ -4581,6 +4586,96 @@
...
+        },
+        "apiVersion": {
+          "type": "string",
+          "description": "The API version."
+        }
+      }
+    },
+    "AliasPathMetadata": {
+      "properties": {
+        "type": {
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4603]: for completeness please mark all readOnly properties as "readOnly": true

Comment (pilor) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4603]: (just doing the new properties is fine)

**Review by pilor — APPROVED** [review]
Minor comments

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/examples/GetNamedProviderAtTenant.json [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+  "parameters": {
+    "$expand": "resourceTypes/aliases",
+    "resourceProviderNamespace": "Microsoft.Storage",
+    "api-version": "2020-06-01"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "namespace": "Microsoft.Storage",
+        "resourceTypes": [
```
Comment (eladperets) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/examples/GetNamedProviderAtTenant.json:11]: This is a partial API response. The full response with the full set of resource types and aliases is huge. LMK if that's a problem

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json [hunk 4]
```diff
@@ -2230,6 +2230,11 @@
         "tags": [
           "Providers"
         ],
+        "x-ms-examples": {
+          "Get a resource provider at tenant scope": {
+            "$ref": "./examples/GetNamedProviderAtTenant.json"
```
Comment (Tiano2017) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:2235]: does the change only apply to tenant level providers call?

**Review by Tiano2017 — APPROVED** [review]

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json (hunk 4 above)
Comment (eladperets) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:2235]: It applies to all providers/ calls. I can add example for each individual flavor if you think it's necessary.

**Comment by eladperets** [comment]
Hi @raych1 what is the next step? Is there anything else i need to do in order to merge this PR?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json [hunk 5]
```diff
@@ -4714,6 +4823,19 @@
           },
           "description": "The API version."
         },
+        "defaultApiVersion": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The default API version."
+        },
+        "apiProfiles": {
```
Comment (raych1) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4831]: I don't see this is being referenced somewhere?

Comment (eladperets) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4831]: what do you mean? it's just a regular property on the response payload, who should be referencing it? see example file

Comment (raych1) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json:4831]: Saw it, thanks.

**Review by raych1 — APPROVED** [review]

**Pull request merged**